src/material.rs

```diff
@@ -1,45 +1,220 @@
+//! Material parameters for mesh rendering.
+
+use color::Color;
 use render::BasicPipelineState;
-use scene::Color;
 use texture::Texture;
 
-/// Material is the enhancement of Texture that is used to setup appearance of [`Mesh`](struct.Mesh.html).
-#[allow(missing_docs)]
+pub use self::basic::Basic;
+
+/// Basic material parameters.
+pub mod basic {
+    use super::*;
+
+    /// Parameters for a basic mesh material.
+    #[derive(Clone, Debug, Default)]
+    pub struct Basic {
```

Comment:
I feel that `basic` module serves little purpose. Would be easier to just move `Pipeline` up

Reply:
The idea is to put the basic pipeline in its own namespace so we can add others without them conflicting. I'll move it up for now.

Reply:
I've removed `Pipeline` in favour of three explicit materials `Basic`, `basic::Custom`, and `Wireframe`.